### ISSUE TITLE
Read-only queue tailer fails on RollCycle due to attempting to write EOF record

### ISSUE THREAD

**Issue by gregallen** [OpenHFT/Chronicle-Queue#527]
exception is

```
java.lang.IllegalStateException: Read Only

	at net.openhft.chronicle.bytes.ReadOnlyMappedBytesStore.checkReadOnly(ReadOnlyMappedBytesStore.java:184)
	at net.openhft.chronicle.bytes.ReadOnlyMappedBytesStore.compareAndSwapInt(ReadOnlyMappedBytesStore.java:45)
	at net.openhft.chronicle.bytes.AbstractBytes.compareAndSwapInt(AbstractBytes.java:168)
	at net.openhft.chronicle.wire.AbstractWire.writeEndOfWire(AbstractWire.java:446)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueStore.writeEOF(SingleChronicleQueueStore.java:349)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueExcerpts$StoreTailer.checkMoveToNextCycle(SingleChronicleQueueExcerpts.java:1046)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueExcerpts$StoreTailer.inACycle(SingleChronicleQueueExcerpts.java:1007)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueExcerpts$StoreTailer.readingDocument(SingleChronicleQueueExcerpts.java:859)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueExcerpts$StoreTailer.readingDocument(SingleChronicleQueueExcerpts.java:812)
	at com.gs.eq.hopper.matching.ingester.ChronicleQueueAdapter.poll(ChronicleQueueAdapter.java:127)
```

**Comment by gregallen** [comment]
Test:

```

import java.io.File;
import java.util.Collections;

import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.TemporaryFolder;

import net.openhft.chronicle.queue.ExcerptTailer;
import net.openhft.chronicle.queue.RollCycles;
import net.openhft.chronicle.queue.impl.single.SingleChronicleQueue;
import net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder;
import net.openhft.chronicle.wire.DocumentContext;
import net.openhft.chronicle.wire.WireType;

public class ChronicleQueueAdapterTest
{
    @Rule
    public TemporaryFolder temporaryFolder = new TemporaryFolder();

    @Test
    public void poll() throws Exception
    {
        final File queuePath = temporaryFolder.newFolder();

        SingleChronicleQueueBuilder.builder(queuePath, WireType.BINARY)
            .rollCycle(RollCycles.TEST_SECONDLY)
            .build()
            .acquireAppender()
            .writingDocument()
            .close();

        final SingleChronicleQueue chronicleQueue = SingleChronicleQueueBuilder.builder(queuePath, WireType.BINARY)
            .rollCycle(RollCycles.TEST_SECONDLY)
            .readOnly(true)
            .build();

        final ExcerptTailer tailer = chronicleQueue.createTailer();
        final String indexPrefix = "myenv-";

        final ChronicleQueueAdapter chronicleQueueAdapter =
            new ChronicleQueueAdapter(tailer, indexPrefix, RollCycles.TEST_SECONDLY, Collections.emptyMap());

        Thread.sleep(1100);

        try (DocumentContext dc = tailer.readingDocument())
        {
            dc.isPresent();
        }
    }
}
```

**Comment by gregallen** [comment]
Perhaps `net.openhft.chronicle.queue.impl.single.SingleChronicleQueueExcerpts.StoreTailer#checkMoveToNextCycle` should not attempt to write EOF if readonly?

**Comment by dpisklov** [comment]
@gregallen Can you ensure you are using latest version and retest? This code doesn't exist in latest queue.
In addition we don't have the ChronicleQueueAdapter code which makes your test not very usable for us...
If this is reproducible with latest queue please feel free to reopen.

**Comment by hft-team-city** [comment]
Released in Chronicle-Queue-5.16.12 (https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.16.12), BOM-1.16.170 (https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-1.16.170)
